phalcon/paginator/adapter/querybuilder.zep

```diff
@@ -68,30 +64,24 @@ class QueryBuilder extends Adapter
 	 */
 	public function __construct(array config)
 	{
-		var builder, limit, page, columns;
-
-		let this->_config = config;
+		var builder, columns;
 
 		if !fetch builder, config["builder"] {
 			throw new Exception("Parameter 'builder' is required");
 		}
 
-		if !fetch limit, config["limit"] {
-			throw new Exception("Parameter 'limit' is required");
-		}
-
 		if fetch columns, config["columns"] {
 		    let this->_columns = columns;
 		}
+		
+		if !isset config["limit"] {
+			throw new Exception("Parameter 'limit' is required");
+		}
```

Comment:
Move this after the `fetch` for the `config["builder"]`. We want to throw an exception as early as possible and avoid extra processing - if the data is incorrect